# Part I - Accident and Emergency Department Synthetic Admission Data Analysis
## by Generoso Roberto

## Introduction

Analysis of health care data has a huge potential to improve health services and patient outcomes but practicing and creating analytical tools and processes is tricky since patient privacy needs to be preserved. The dataset that is used is the A&E Synthetic Data which created as a result of an NHS England pilot project to share more data without sacrifising the privacy of patients. The project used statistical models to that tries create datasets that retains the value of the original data but can not be used to re-identify specific individual records. 

This provides an opportunity for us to get to look and analyze the A&E data that NHS collects. For this analysis the focus is on admitted cases in just one hospital.

The columns in the dataset are the following:
* `IMD_Decile_From_LSOA` - measure of deprivation. A score of 1 indicates the entry comes from the 10% most deprived area in England
* `Age_Band` - The interval of the age of the patient
* `Sex` - Coded in 1 and 2 to represent the two sexes
* `AE_Arrive_Date` - The date the patient arrived at the A&E provider
* `AE_Arrive_HourOfDay` - The interval of what time the patient arrived at the A&E
* `AE_Time_Mins` - Total number of minutes the patient stayed in the A&E
* `AE_HRG` - Code for the health reference group
* `AE_Num_Diagnoses` - Number of diagnoses the patient have in the A&E
* `AE_Num_Investigations` - Number of investigations done to the patient
* `AE_Num_Treatments` - Number of treatments done to the patient
* `AE_Arrival_Mode` - Code for the mode by which  patient arrived
* `Provider_Patient_Distance_Miles` - Distance from the patient place of origin to the provider in miles
* `Admission_Method` - Code for the method how the patient was admitted
* `ICD10_Chapter_Code` - International classification of disease chapter code
* `Treatment_Function_Code` - The treatment function under which the patient is treated.
* `Length_Of_Stay_Days` - The number of days the patient was admitted

## Preliminary Data Wrangling

In [1]:
# import all packages and set plots to be embedded inline
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb

%matplotlib inline

In [2]:
#Load the dataset
aedata_df=pd.read_csv('AnE_Synthetic_Data_Provider_Admission.csv')

In [5]:
# High-level overview of data shape and composition
print(aedata_df.shape)
print(aedata_df.info())
print(aedata_df.head(10))

(326444, 16)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 326444 entries, 0 to 326443
Data columns (total 16 columns):
 #   Column                           Non-Null Count   Dtype  
---  ------                           --------------   -----  
 0   IMD_Decile_From_LSOA             326444 non-null  int64  
 1   Age_Band                         326444 non-null  object 
 2   Sex                              326444 non-null  int64  
 3   AE_Arrive_Date                   326444 non-null  object 
 4   AE_Arrive_HourOfDay              326444 non-null  object 
 5   AE_Time_Mins                     326444 non-null  float64
 6   AE_HRG                           326372 non-null  object 
 7   AE_Num_Diagnoses                 326444 non-null  int64  
 8   AE_Num_Investigations            326444 non-null  int64  
 9   AE_Num_Treatments                326444 non-null  int64  
 10  AE_Arrival_Mode                  326444 non-null  int64  
 11  Provider_Patient_Distance_Miles  326444 non-null  fl

1. There are 326444 rows in the data
2. `AE_Arrive_Date` is an object and should be made into a datetime
3. The following columns are ordered categorical types
    * `Age_Band`
    * `AE_Arrive_HourOfDay`
    * `AE_HRG`
4. The following columns have null values:
    * `AE_HRG`
    * `ICD10_Chapter_Code`
    * `Treatment_Function_Code`
    * `Length_Of_Stay_Days`

### What is the structure of your dataset?

> Your answer here!

### What is/are the main feature(s) of interest in your dataset?

> Your answer here!

### What features in the dataset do you think will help support your investigation into your feature(s) of interest?

> Your answer here!

## Univariate Exploration

> In this section, investigate distributions of individual variables. If
you see unusual points or outliers, take a deeper look to clean things up
and prepare yourself to look at relationships between variables.


> **Rubric Tip**: The project (Parts I alone) should have at least 15 visualizations distributed over univariate, bivariate, and multivariate plots to explore many relationships in the data set.  Use reasoning to justify the flow of the exploration.



>**Rubric Tip**: Use the "Question-Visualization-Observations" framework  throughout the exploration. This framework involves **asking a question from the data, creating a visualization to find answers, and then recording observations after each visualisation.** 




>**Rubric Tip**: Visualizations should depict the data appropriately so that the plots are easily interpretable. You should choose an appropriate plot type, data encodings, and formatting as needed. The formatting may include setting/adding the title, labels, legend, and comments. Also, do not overplot or incorrectly plot ordinal data.

### Discuss the distribution(s) of your variable(s) of interest. Were there any unusual points? Did you need to perform any transformations?

> Your answer here!

### Of the features you investigated, were there any unusual distributions? Did you perform any operations on the data to tidy, adjust, or change the form of the data? If so, why did you do this?

> Your answer here!

## Bivariate Exploration

> In this section, investigate relationships between pairs of variables in your
data. Make sure the variables that you cover here have been introduced in some
fashion in the previous section (univariate exploration).

### Talk about some of the relationships you observed in this part of the investigation. How did the feature(s) of interest vary with other features in the dataset?

> Your answer here!

### Did you observe any interesting relationships between the other features (not the main feature(s) of interest)?

> Your answer here!

## Multivariate Exploration

> Create plots of three or more variables to investigate your data even
further. Make sure that your investigations are justified, and follow from
your work in the previous sections.

### Talk about some of the relationships you observed in this part of the investigation. Were there features that strengthened each other in terms of looking at your feature(s) of interest?

> Your answer here!

### Were there any interesting or surprising interactions between features?

> Your answer here!

## Conclusions
>You can write a summary of the main findings and reflect on the steps taken during the data exploration.



> Remove all Tips mentioned above, before you convert this notebook to PDF/HTML


> At the end of your report, make sure that you export the notebook as an
html file from the `File > Download as... > HTML or PDF` menu. Make sure you keep
track of where the exported file goes, so you can put it in the same folder
as this notebook for project submission. Also, make sure you remove all of
the quote-formatted guide notes like this one before you finish your report!

